### ISSUE TITLE
Sync stall

### ISSUE THREAD

**Issue by greensea** [openethereum/openethereum#11366]
- **Parity Ethereum version**: v2.6.8-beta-9bf6ed8-20191231/x86_64-linux-gnu/rustc1.40.0
- **Operating system**:  Linux
- **Installation**: Binary from GitHub release page
- **Fully synchronized**:  yes
- **Network**: ethereum
- **Restarted**: yes

Parity client stall after running for about 1 days, it stuck at a block. If I restart it, it can continue to sync.
This is the output when this issue happens:
```
2020-01-04 20:11:38  Snapshot: 2643970 accounts, 5000 blocks, 578130870 bytes
2020-01-04 20:11:42  Imported #9213796 0xaecf…c888 (75 txs, 3.90 Mgas, 778 ms, 15.57 KiB)
2020-01-04 20:11:45  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:11:55  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:12:01    16/25 peers   4 MiB chain 18 MiB db 0 bytes queue 736 KiB sync  RPC:  0 conn,    1 req/s,  248 µs
2020-01-04 20:12:05  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:12:15  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:12:25  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:12:31  Imported #9213797 0x1a6b…d82d (0 txs, 0.00 Mgas, 110 ms, 0.53 KiB)
2020-01-04 20:12:36    13/25 peers   1014 KiB chain 18 MiB db 0 bytes queue 883 KiB sync  RPC:  0 conn,    3 req/s,   18 µs
2020-01-04 20:12:52  Local tx 0x292bb5d4677edf1f81117b54bcf792751d5c4bd1563c0278bd340164d8ff7230 below minimal gas price accepted
2020-01-04 20:12:53  Imported #9213803 0x7126…f95c (3 txs, 0.07 Mgas, 8987 ms, 0.91 KiB) + another 5 block(s) containing 495 tx(s)
2020-01-04 20:12:55  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:08  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:08    12/25 peers   3 MiB chain 18 MiB db 0 bytes queue 704 KiB sync  RPC:  0 conn,    3 req/s,   17 µs
2020-01-04 20:13:09  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:13  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:15  Imported #9213806 0xe281…3fcf (7 txs, 0.27 Mgas, 1806 ms, 1.61 KiB) + another 2 block(s) containing 121 tx(s)
2020-01-04 20:13:16  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:28  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:35  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:36  Syncing #9213806 0xe281…3fcf     0.11 blk/s    4.7 tx/s    0.2 Mgas/s      0+    0 Qed  #9213796   11/25 peers   3 MiB chain 18 MiB db 0 bytes queue 738 KiB sync  RPC:   0 conn,    2 req/s,   17 µs
2020-01-04 20:13:41  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed  #9197410   12/25 peers   3 MiB chain 18 MiB db 0 bytes queue 738 KiB sync  RPC:   0 conn,    2 req/s,   19 µs
2020-01-04 20:13:45  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:13:46  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed  #7116642    8/25 peers   7 MiB chain 18 MiB db 0 bytes queue 738 KiB sync  RPC:   0 conn,    3 req/s,   17 µs
2020-01-04 20:13:55  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:01  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed        #0    7/25 peers   6 MiB chain 18 MiB db 0 bytes queue 737 KiB sync  RPC:   0 conn,    3 req/s,   16 µs
2020-01-04 20:14:05  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:15  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:25  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:31     6/25 peers   6 MiB chain 18 MiB db 0 bytes queue 813 KiB sync  RPC:  0 conn,    1 req/s,   16 µs
2020-01-04 20:14:35  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:41  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed        #0    6/25 peers   2 MiB chain 18 MiB db 0 bytes queue 821 KiB sync  RPC:   0 conn,    0 req/s,   16 µs
2020-01-04 20:14:41  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed        #0    7/25 peers   7 MiB chain 18 MiB db 0 bytes queue 821 KiB sync  RPC:   0 conn,    0 req/s,   16 µs
2020-01-04 20:14:45  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:47  Local tx 0x26f19d37f028a528d92a2151d8621799e317e92f1c962a093f2b67c35289435e below minimal gas price accepted
2020-01-04 20:14:52  Local tx 0xddf621e892e16e35bc7073f688f470d04ccb6fd488932e81ebd0a9800eaf3f6e below minimal gas price accepted
2020-01-04 20:14:55  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:14:55  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed        #2    7/25 peers   2 MiB chain 18 MiB db 0 bytes queue 921 KiB sync  RPC:   0 conn,    1 req/s,   16 µs
2020-01-04 20:14:55  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed        #2    7/25 peers   4 MiB chain 18 MiB db 0 bytes queue 921 KiB sync  RPC:   0 conn,    1 req/s,   16 µs
2020-01-04 20:14:56  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed        #2    7/25 peers   4 MiB chain 18 MiB db 0 bytes queue 922 KiB sync  RPC:   0 conn,    1 req/s,   16 µs
2020-01-04 20:15:03  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #255    7/25 peers   4 MiB chain 18 MiB db 0 bytes queue 824 KiB sync  RPC:   0 conn,    0 req/s,   16 µs
2020-01-04 20:15:05  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:15:08  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #255    7/25 peers   4 MiB chain 18 MiB db 0 bytes queue 955 KiB sync  RPC:   0 conn,    2 req/s,   16 µs
2020-01-04 20:15:11  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #255    7/25 peers   4 MiB chain 18 MiB db 0 bytes queue 1012 KiB sync  RPC::  0 conn,    1 req/s,   16 µs
2020-01-04 20:15:15  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:15:16  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #508    7/25 peers   4 MiB chain 18 MiB db 0 bytes queue 782 KiB sync  RPC:   0 conn,    0 req/s,   16 µs
2020-01-04 20:15:23  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #638    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 762 KiB sync  RPC:   0 conn,    3 req/s,   17 µs
2020-01-04 20:15:25  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:15:31  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #763    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 771 KiB sync  RPC:   0 conn,    2 req/s,   17 µs
2020-01-04 20:15:31  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #763    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 771 KiB sync  RPC:   0 conn,    2 req/s,   17 µs
2020-01-04 20:15:35  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:15:36  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #889    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 834 KiB sync  RPC:   0 conn,    1 req/s,   17 µs
2020-01-04 20:15:41  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #896    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 1001 KiB sync  RPC::  0 conn,    2 req/s,   17 µs
2020-01-04 20:15:45  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:15:46  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #896    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 1 MiB sync  RPC:  00 conn,    3 req/s,   17 µs
2020-01-04 20:15:51  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed      #896    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 1 MiB sync  RPC:  00 conn,    3 req/s,   17 µs
2020-01-04 20:15:55  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
2020-01-04 20:15:58  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed     #1283    5/25 peers   4 MiB chain 18 MiB db 0 bytes queue 731 KiB sync  RPC:   0 conn,    1 req/s,   17 µs
2020-01-04 20:16:01  Syncing #9213806 0xe281…3fcf     0.00 blk/s    0.0 tx/s    0.0 Mgas/s      0+    0 Qed     #1397    4/25 peers   4 MiB chain 18 MiB db 0 bytes queue 588 KiB sync  RPC:   0 conn,    1 req/s,   17 µs
2020-01-04 20:16:05  Snapshot: 2679111 accounts, 5000 blocks, 581367911 bytes
```

Expected behavior:
Parity should not stall on syncing.

Setps to reproduce:
There is no specific steps to reproduce, I just run parity very normaly. The command line is:
```
parity --tracing=on --pruning=archive --db-path=/data/ethereum --jsonrpc-interface=all
```

**Comment by joshua-mir** [comment]
this stall is occurring due to snapshot generation - are you seeing resources running out?

unfortunately this is part of the compromise we had to make to maintain network reliability and in some cases it reduces node reliability. you can force this to be disabled with `--no-periodic-snapshots`

**Comment by greensea** [comment]
My server disk is busy, but it should have enough resource, parity is able to sync to latest block at most of the time.
I try add --no-periodic-snapshots parameter, but this issue still exists. But there is some different, without this parameter, sync stall in one or two days, after adding this parameter, sync stall happens after about a week. 
Anyway, after adding --no-periodic-snapshots parameter, sync stall still happen, but not so frequently. Is there any possible reason cause sync stall?

**Comment by dvdplm** [comment]
It is possible that this is fixed in 2.7.2; have you tried the latest version? Be aware that upgrading is a one-way thing: you cannot revert to 2.6.8 so check your backups. :)
